apps/stats/views.py

```diff
@@ -699,3 +705,73 @@ def render_json(request, addon, stats):
     fudge_headers(response, stats)
     json.dump(stats, response, cls=DecimalJSONEncoder)
     return response
+
+
+class ArchiveMixin(object):
+    """Provides common helper methods for all archive views."""
+    def get_addon(self, request, addon_id):
+        """Fetches an addon by `id` or `slug`"""
```

Comment:
I'm not sure we want both, do we? Let's just keep it simple if possible and only support one.